### ISSUE TITLE
Setting the `Module['websocket']` at runtime does not work since version 3.1.71

### ISSUE THREAD

**Issue by jerome-caucat** [emscripten-core/emscripten#22969]
Since version 3.1.71, setting the `Module['websocket']` at runtime does not work anymore.

I tried setting `Module['websocket']` using `emscripten_run_script` and `EM_JS`.

I think it could be linked to https://github.com/emscripten-core/emscripten/pull/22774.

**Version of emscripten/emsdk:**

```
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 3.1.72 (437140d149d9c977ffc8b09dbaf9b0f5a02db190)
clang version 20.0.0git (https:/github.com/llvm/llvm-project 1c4caece05f1885ba6ed80755d6b5de1b9f99579)
Target: wasm32-unknown-emscripten
Thread model: posix
InstalledDir: /emsdk/upstream/bin
```

**Build command:**

```
emcc test.cpp -o index.html -sEXPORTED_FUNCTIONS=_main -sSOCKET_DEBUG=1
```

**Source test.cpp:**

```cpp
#include <emscripten/emscripten.h>
#include <errno.h>
#include <iostream>
#include <arpa/inet.h>
#include <netinet/in.h>
#include <sys/socket.h>
#include <string.h>

int main() {
    emscripten_run_script("Module['websocket'] = {url: 'wss://MY_URL', subprotocol: 'binary'}");

    int _fd = ::socket(PF_INET, SOCK_STREAM, IPPROTO_TCP);
    sockaddr_in server;
    memset(&server, 0, sizeof(server));
    server.sin_addr.s_addr = inet_addr("172.29.1.0");
    server.sin_family = AF_INET;
    server.sin_port = htons(443);

    auto res = ::connect(_fd, (struct sockaddr*)&server , sizeof(server));
    std::cout << "!! " << res << "  " << errno << std::endl;

    return 0;
}
```

**Result with 3.1.72:**

```
websocket: connect: ws://172.29.1.0:443/, binary
!! -1  23
```

**Result with 3.1.72, using the `emcc` option `-sWEBSOCKET_URL=wss://MY_URL`:**

```
websocket: connect: wss://MY_URL, binary
websocket: adding peer: 172.29.1.0:443
!! 0  0
```

**Result with 3.1.70:**

```
websocket: connect: wss://MY_URL, binary
websocket: adding peer: 172.29.1.0:443
!! 0  0
```

**Comment by sbc100** [comment]
Ah yes, that is likely due to #22774.

Normally with `Module` properties like this one you are supposed to set them before the module is loaded.   See https://emscripten.org/docs/api_reference/module.html#module-object and https://emscripten.org/docs/api_reference/module.html#compilation-settings and https://emscripten.org/docs/tools_reference/settings_reference.html?highlight=environment#incoming-module-js-api

Would it be possible to supply this setting in advance in your use case?   I'd rather not revert #22774.

**Comment by jerome-caucat** [comment]
> Would it be possible to supply this setting in advance in your use case?

Hum, not really, `MY_URL` is in the form `wss://DOMAIN/PATH/ID?ARGS`, `ID` and `ARGS` are computed at runtime.

Not sure if I can leverage the [prefix-only option](https://emscripten.org/docs/tools_reference/settings_reference.html?highlight=environment#websocket-url).

**Comment by sbc100** [comment]
> ```c++
> emscripten_run_script
> ```

I that case I think maybe we should create a new runtime API for setting this stuff?   I don't think allowing ` emscripten_run_script("Module['websocket']  ..` after the module has been created was ever intended behaviour (But I could be wrong).

In the mean time you it looks like you could assign to the new SOCKS.websocketArgs variable instead.  e.g. `emscripten_run_script("SOCKFS.websocketArgs = {'url': ..., 'protocol': ...}")`?  Does that work?

**Comment by jerome-caucat** [comment]
> In the mean time you it looks like you could assign to the new SOCKS.websocketArgs variable instead. e.g. `emscripten_run_script("SOCKFS.websocketArgs = {'url': ..., 'subprotocol': ...}")`? Does that work?

Yes! Thank you very much for this workaround. That will do the job until a dedicated feature is added.
